src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -14,31 +14,45 @@ import {IStrategyManager} from "lib/eigenlayer-contracts/src/contracts/interface
 import {IDelegationManager, IDelegationManagerTypes} from "lib/eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
 import {IStrategy} from "lib/eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
 import {IRewardsCoordinator} from "lib/eigenlayer-contracts/src/contracts/interfaces/IRewardsCoordinator.sol";
+import {IAllocationManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IAllocationManager.sol";
 import {ITokenStakingNode} from "src/interfaces/ITokenStakingNode.sol";
 import {ITokenStakingNodesManager} from "src/interfaces/ITokenStakingNodesManager.sol";
 import {IWrapper} from "src/interfaces/IWrapper.sol";
 import {IYieldNestStrategyManager} from "src/interfaces/IYieldNestStrategyManager.sol";
-import {IDelegationManagerExtended} from "src/external/eigenlayer/IDelegationManagerExtended.sol";
+import {DelegationManagerStorage} from "lib/eigenlayer-contracts/src/contracts/core/DelegationManagerStorage.sol";
+import {IAssetRegistry} from "src/interfaces/IAssetRegistry.sol";
+import {IynEigen} from "src/interfaces/IynEigen.sol";
 
 interface ITokenStakingNodeEvents {
     event DepositToEigenlayer(IERC20 indexed asset, IStrategy indexed strategy, uint256 amount, uint256 eigenShares);
     event Delegated(address indexed operator, bytes32 approverSalt);
     event Undelegated(bytes32[] withdrawalRoots);
     event QueuedWithdrawals(IStrategy strategies, uint256 shares, bytes32[] fullWithdrawalRoots);
-    event CompletedQueuedWithdrawals(uint256 shares, uint256 amountOut, address strategy);
     event DeallocatedTokens(uint256 amount, IERC20 token);
     event CompletedManyQueuedWithdrawals(IDelegationManager.Withdrawal[] withdrawals);
     event ClaimerSet(address indexed claimer);
+    event QueuedSharesSynced();
 }
 
+// /// This interface is created because the src/interfaces/IynEigen.sol does not provide a way to obtain the assetRegistry.
+// /// TODO: Should we expose assetRegistry in IynEigen?
```

Comment:
can we address these todos and remove commented out code?

Reply:
Fixed at 99854cf7cab54156e7d0e1e85870f7cdb75cbb24